addons/openebs/3.3.0/install.sh

```diff
@@ -48,12 +48,58 @@ function openebs() {
 function openebs_maybe_migrate_from_rook() {
     if [ -z "$ROOK_VERSION" ]; then
         if kubectl get ns | grep -q rook-ceph; then
+            # show validation errors from pvmigrate
+            # if there are errors, maybe_prompt_migrate_from_rook_dryrun_errors() will bail
+            maybe_prompt_migrate_from_rook_dryrun_errors
             rook_ceph_to_sc_migration "$OPENEBS_LOCALPV_STORAGE_CLASS"
             DID_MIGRATE_ROOK_PVCS=1 # used to automatically delete rook-ceph if object store data was also migrated
         fi
     fi
 }
 
+function maybe_prompt_migrate_from_rook_dryrun_errors() {
+    echo "Running Rook to OpenEBS Migration Checks ..."
+
+    # get the list of StorageClasses that use rook-ceph
+    rook_scs=$(kubectl get storageclass | grep rook | grep -v '(default)' | awk '{ print $1}') # any non-default rook StorageClasses
+    rook_default_sc=$(kubectl get storageclass | grep rook | grep '(default)' | awk '{ print $1}') # any default rook StorageClasses
+
+    local rook_scs_pvmigrate_dryrun_output
+    local rook_default_sc_pvmigrate_dryrun_output
+    for rook_sc in $rook_scs
+    do
+        # run validation checks for non default Rook storage classes
+        rook_scs_pvmigrate_dryrun_output=$($BIN_PVMIGRATE --source-sc "$rook_sc" --dest-sc "$OPENEBS_LOCALPV_STORAGE_CLASS" --dry-run 2>/dev/null)
+    done
+
+    if [ -n "$rook_default_sc" ] ; then
+        # run validation checks for Rook default storage class
+        rook_default_sc_pvmigrate_dryrun_output=$($BIN_PVMIGRATE --source-sc "$rook_default_sc" --dest-sc "$OPENEBS_LOCALPV_STORAGE_CLASS" --dry-run 2>/dev/null)
+    fi
+
+    if { [ -n "$rook_scs_pvmigrate_dryrun_output" ] && [[ "$rook_scs_pvmigrate_dryrun_output" != *"ProvisioningFailed"* ]]; } || { [ -n "$rook_default_sc_pvmigrate_dryrun_output" ] && [[ "$rook_default_sc_pvmigrate_dryrun_output" != *"ProvisioningFailed"* ]]; }; then
+        printf "${YELLOW}"
```

Comment:
Could we create a lib/add to an utils/common file functions such as the following example? So that, every time that we want print a waring that will be yellow, an error will be red and an info blue for example. Also, that can avoid we per mistake forget to reset the color. 

See: 

**Lib example**

```shell
# Turn colors in this script off by setting the NO_COLOR variable in your
# environment to any value:
NO_COLOR=${NO_COLOR:-""}
if [ -z "$NO_COLOR" ]; then
  warning_color=$'\e[1;33m' # required check what would be yellow here
  error_color=$'\e[0;31m'
  reset_color=$'\e[0m'
else
  warning_color=''
  error_color=''
  reset_color=''
fi

function log() { printf '%s\n' "$*"; }
function error() { error_text "ERROR:" $* >&2; }
function fatal() { error "$@"; exit 1; }

function warning_text {
  echo "$warning_color$*$reset_color"
}

function error_text {
  echo "$error_color$*$reset_color"
}

```

**Usage example**

```
 echo -e "my test for the error :\n${value}"

OR 

warning_text "My warning text will be outputted in yellow"
```

Reply:
@rrpolanco after look it better see that the kURL has an standard/utils to print the warnings already (i.e. https://github.com/replicatedhq/kURL/blob/d8de81118f3a113036d3125733ce806d3421e8c1/addons/rook/1.5.11/install.sh#L43 )

Look:

<img width="584" alt="Screenshot 2022-11-28 at 18 55 42" src="https://user-images.githubusercontent.com/7708031/204358548-bbdf8647-e3fb-4e1e-a0c8-c258c4d9fbc7.png">

WDYT about we try to keep here all using the same approaches? So that if we want to improve or change that can be done horizontally.

Reply:
Good suggestion. I'll use `logFail()` to output the validation failures from the command.